include/Graph/Graph.hpp

```diff
@@ -1449,40 +1449,31 @@ namespace CXXGRAPH
 			}
 		}
 
-		// An array to store index of the cheapest edge of
-		// subset.  stored index refers to the edgeId
-		std::vector<long> cheapest(n, -1);
-		// user can give arbitrary ids to nodes
-		// we map these ids from 0 to 1 for consistency
-		// NOTE: WE CAN REMOVE THIS WHEN WE TAKE CARE OF THIS GLOBALLY
-		// WHILE CONSTRUCTING THE GRAPH
-		std::map<unsigned long long, unsigned long long> userNodeMap;
-		unsigned long long i = 0;
 		for (auto node : nodeSet)
 		{
-			userNodeMap[node->getId()] = i;
-			Subset set{i, 0};
-			subsets.push_back(set);
-			i++;
+			Subset set{node->getId(), 0};
+			subsets[node->getId()] = set;
 		}
 
 		result.mstCost = 0; //we will store the cost here
 		//exit when only 1 tree i.e. mst
 		while (numTrees > 1)
 		{
-			// Everytime initialize cheapest array
-			std::fill(cheapest.begin(), cheapest.end(), -1);
+			// Everytime initialize cheapest map
+			// It stores index of the cheapest edge of subset.
+			std::map<unsigned long long, unsigned long long> cheapest;
+			for (auto node : nodeSet)
```

Comment:
This range for loop should be used like    _"for (const auto& node: nodeSet"_  since:
- We do not need local copies of each element that we iterate through, so saving computational cost.
- We are not changing any of the elements in the iterator, so it can be defined as "const". So whoever that reads this code knows that in this loop, out iterable structure is not modified.

I propose this improvement to be done in a new ticket since I saw it in several places.

Reply:
yes, I agree.
In this moment I approve this pull request, the you can add a new ticket